piksi_multi_cpp/src/observations/udp_observation_receiver.cpp

```diff
@@ -0,0 +1,80 @@
+#include <piksi_multi_cpp/observations/udp_observation_receiver.h>
+// udp
+#include <arpa/inet.h>
+#include <fcntl.h>  // File control definitions
+#include <netdb.h>
+#include <netinet/in.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>  // UNIX standard function definitions
+
+namespace piksi_multi_cpp {
+UDPObservationReceiver::UDPObservationReceiver(
+    ObservationsConsumer::Ptr consumer)
```

Comment:
[I'd also pass shared_ptr as const reference ](https://stackoverflow.com/questions/3310737/should-we-pass-a-shared-ptr-by-reference-or-by-value)

Reply:
hm good point ( I agree).

Clang-tidy even suggest using pass-by-value and use move sematics in newer c++ versions.

Would look like this
`  UDPObservationReceiver::UDPObservationReceiver(
    ObservationsConsumer::Ptr consumer)
      : : consumers_({std::move(consumer)})  {}
`

not yet sure which and why is the best :-)